### ISSUE TITLE
Accessibility bug: combo boxes (on checkout page, etc) are inaccessible to screen readers

### ISSUE THREAD

**Issue by lilmike** [woocommerce/woocommerce#14683]
### EXPLANATION OF THE ISSUE

Woocommerce 3.0.4: When using a screen reader (I've tested NVDA and Jaws for Windows on windows), the combo boxes (for example to select a state, country, etc on the checkout page) are inaccessible. When arrowing through the combo box, it simply reads nothing, and when using the read current line keystroke, it reads blank.

### STEPS TO REPRODUCE THE ISSUE

1. Go to a WooCommerce enabled store.
2. Add a product to the cart, and proceed to checkout.
3. Open a screen reader (I recommend NVDA because it is free to use), and try to access the state or country combo box. You will notice the simptomes above.

### SYSTEM STATUS

<details>
```
WordPress environment
Home URL:
 
https://2mb.solutions
Site URL:
 
https://2mb.solutions
WC version:
 
3.0.4
Log directory writable:
 
 /storage/websites/2mb.solutions/public_html/wp-content/uploads/wc-logs/
WP version:
 
4.7.4
WP multisite:
  
WP memory limit:
 
512 MB
WP debug mode:
 
–
WP cron:
  
Language:
 
en_US
Server environment
Server info:
 
nginx/1.12.0
PHP version:
 
7.1.4
PHP post max size:
 
64 MB
PHP time limit:
 
300
PHP max input vars:
 
1000
cURL version:
 
7.54.0, OpenSSL/1.1.0e
SUHOSIN installed:
 
–
Max upload size:
 
1 MB
Default timezone is UTC:
  
fsockopen/cURL:
  
SoapClient:
 
 Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument:
  
GZip:
  
Multibyte string:
  
Remote post:
  
Remote get:
  
Database
WC database version:
 
3.0.4
Database prefix
 
wp2mb_
woocommerce_sessions
  
woocommerce_api_keys
  
woocommerce_attribute_taxonomies
  
woocommerce_downloadable_product_permissions
  
woocommerce_order_items
  
woocommerce_order_itemmeta
  
woocommerce_tax_rates
  
woocommerce_tax_rate_locations
  
woocommerce_shipping_zones
  
woocommerce_shipping_zone_locations
  
woocommerce_shipping_zone_methods
  
woocommerce_payment_tokens
  
woocommerce_payment_tokenmeta
  
MaxMind GeoIP database:
 
 /storage/websites/2mb.solutions/public_html/wp-content/uploads/GeoIP.dat
Security
Secure connection (HTTPS):
  
Hide errors from visitors
  
Active plugins (39)
Visit plugin homepage
 
by Nocean – 1.10
Visit plugin homepage
 
by Takayuki Miyoshi – 4.7
Visit plugin homepage
 
by Cryout Creations – 0.5.7
Disable WooCommerce Phone Requirement and make free checkout easier
 
by 2MB Solutions – 1.0
Visit plugin homepage
 
by Elegant Themes – 1.0
Visit plugin homepage
 
by Automattic – 4.8.2
Visit plugin homepage
 
by Vladimir Kolesnikov – 0.2.5
Product Open Pricing (Name Your Price) for WooCommerce
 
by Algoritmika Ltd – 1.0.0
Visit plugin homepage
 
by WPMU DEV – 3.0.4
Visit plugin homepage
 
by socialmediafeather – 1.8.3
Visit plugin homepage
 
by Pino Ceniccola – 0.2.1.2
Visit plugin homepage
 
by 2MB Solutions – 0.0.1
Visit plugin homepage
 
by WooCommerce – 3.1.7
Visit plugin homepage
 
by Automattic – 3.0.4
Visit plugin homepage
 
by Wordfence – 6.3.6
Visit plugin homepage
 
by wokamoto – 1.1.7
Visit plugin homepage
 
by WPMU DEV – 1.3
Visit plugin homepage
 
by WPMU DEV – 1.4.2
Visit plugin homepage
 
by David Anderson, Ruhani Rabin, Team Updraft – 2.1.1
Visit plugin homepage
 
by iControlWP – 5.8.2
Visit plugin homepage
 
by WPMU DEV – 2.6.2
Visit plugin homepage
 
by WPMU DEV – 4.3
Visit plugin homepage
 
by WPMU DEV – 4.4.2.4
Visit plugin homepage
 
by WPMU DEV – 4.3
Visit plugin homepage
 
by Automattic – 4.8.2
Visit plugin homepage
 
by Vladimir Kolesnikov – 0.2.5
Visit plugin homepage
 
by iControlWP – 5.8.2
Visit plugin homepage
 
by WPMU DEV – 2.6.2
Visit plugin homepage
 
by WPMU DEV – 1.4.2
Visit plugin homepage
 
by WPMU DEV – 1.3
Visit plugin homepage
 
by Wordfence – 6.3.6
Visit plugin homepage
 
by WPMU DEV – 3.0.4
Visit plugin homepage
 
by rtCamp – 1.9.10
Visit plugin homepage
 
by WangGuard – 1.7.3
Visit plugin homepage
 
by WebSharks, Inc. – 170220
Visit plugin homepage
 
by wpUXsolutions – 2.4.4
Visit plugin homepage
 
by Postmatic – 1.0.14
Visit plugin homepage
 
by Daniele De Santis – 1.0.1
Visit plugin homepage
 
by WPMU DEV – 2.0
Settings
API enabled:
  
Force SSL:
 
–
Currency
 
USD ($)
Currency position
 
left
Thousand separator
 
,
Decimal separator
 
.
Number of decimals
 
2
Taxonomies: Product types 
 
external (external), grouped (grouped), simple (simple), variable (variable)
Taxonomies: Product visibility 
 
exclude-from-catalog (exclude-from-catalog), exclude-from-search (exclude-from-search), featured (featured), outofstock (outofstock), rated-1 (rated-1), rated-2 (rated-2), rated-3 (rated-3), rated-4 (rated-4), rated-5 (rated-5)
WC pages
Edit Shop base page:
 
#159 - /shop/
Edit Cart page:
 
#160 - /cart/
Edit Checkout page:
 
#161 - /checkout/
Edit My account page:
 
#162 - /my-account/
Theme
Name:
 
Parabola
Version:
 
2.1.3
Author URL:
 
https://www.cryoutcreations.eu
Child theme:
 
 – If you're modifying WooCommerce on a parent theme you didn't build personally, then we recommend using a child theme. See: How to create a child theme
WooCommerce support:
  
Templates
Archive template:
 
Your theme has a woocommerce.php file, you will not be able to override the woocommerce/archive-product.php custom template since woocommerce.php has priority over archive-product.php. This is intended to prevent display issues.
Overrides:
 
–
```
</details>

**Comment by mikejolley** [comment]
We're aware and this has been noted in: https://github.com/woocommerce/woocommerce/issues/14317

Select2 is the script which lacks screen reader support. We need to decide to fork or patch, but this is not a short term fixable thing.

In the meantime, it can be dequeued and replaced, or left as the default select box.

**Comment by lilmike** [comment]
Hi,
Thanks for letting me know. When you say left as the default select box, do you mean the standard web browser select box? If so, how can I do that?
Thanks,
Michael.

Sent from my iPhone

On Apr 25, 2017, at 8:10 AM, Mike Jolley <notifications@github.com<mailto:notifications@github.com>> wrote:


We're aware and this has been noted in: #14317<https://github.com/woocommerce/woocommerce/issues/14317>

Select2 is the script which lacks screen reader support. We need to decide to fork or patch, but this is not a short term fixable thing.

In the meantime, it can be dequeued and replaced, or left as the default select box.

**Comment by mikejolley** [comment]
```
add_action( 'wp_enqueue_scripts', 'remove_select2', 11 );

function remove_select2() {
	wp_dequeue_script( 'select2' );
}
```
